src/lib/userAnalytics/utils.ts

```diff
@@ -0,0 +1,208 @@
+
+  if (!metricData) {
+    metrics.pushError("Order metric data not found", "sendUserAnalyticsTradeBoxConfirmClickEvent");
+    return;
+  }
+
+  if (metricData.metricType === "increasePosition") {
+    userAnalytics.pushEvent<TradeBoxConfirmClickEvent>({
+      event: "TradeBoxAction",
+      data: {
+        action: "IncreasePositionConfirmClick",
+        pair: metricData.marketIndexName || "",
+        pool: metricData.marketPoolName || "",
+        type: metricData.isLong ? "Long" : "Short",
+        orderType: metricData.orderType === OrderType.MarketIncrease ? "Market" : "Limit",
+        tradeType: metricData.hasExistingPosition ? "IncreaseSize" : "InitialTrade",
+        leverage: metricData.leverage || "",
+        is1CT: metricData.is1ct,
+        chain: getChainName(chainId),
+        isFirstOrder: Boolean(metricData.isFirstOrder),
+      },
+    });
+  } else if (metricData.metricType === "decreasePosition") {
+    userAnalytics.pushEvent<TradeBoxConfirmClickEvent>({
+      event: "TradeBoxAction",
+      data: {
+        action: "DecreasePositionConfirmClick",
+        pair: metricData.marketIndexName || "",
+        pool: metricData.marketPoolName || "",
+        type: metricData.isLong ? "Long" : "Short",
+        orderType: metricData.orderType === OrderType.MarketDecrease ? "Market" : "TPSL",
+        tradeType: metricData.isFullClose ? "ClosePosition" : "DecreaseSize",
+        leverage: "",
+        is1CT: metricData.is1ct,
+        chain: getChainName(chainId),
+        isFirstOrder: false,
+      },
+    });
+  } else if (metricData.metricType === "swap") {
+    userAnalytics.pushEvent<TradeBoxConfirmClickEvent>({
+      event: "TradeBoxAction",
+      data: {
+        action: "SwapConfirmClick",
+        pair: `${metricData.initialCollateralSymbol}/${metricData.toTokenSymbol}`,
+        pool: "",
+        type: "Swap",
+        orderType: metricData.orderType === OrderType.MarketSwap ? "Market" : "Limit",
+        tradeType: "InitialTrade",
+        leverage: "",
+        is1CT: metricData.is1ct,
+        chain: getChainName(chainId),
+        isFirstOrder: false,
+      },
+    });
+  } else if (metricData.metricType === "buyGM") {
+    userAnalytics.pushEvent<PoolsPageBuyConfirmEvent>({
+      event: "PoolsPageAction",
+      data: {
+        action: "BuyConfirm",
+        type: "GM",
+        poolName: metricData.marketName || "",
+        glvAddress: "",
+        amount: metricData.marketTokenUsd || 0,
+        isFirstBuy: Boolean(metricData.isFirstBuy),
+      },
+    });
+  } else if (metricData.metricType === "buyGLV") {
+    userAnalytics.pushEvent<PoolsPageBuyConfirmEvent>({
+      event: "PoolsPageAction",
+      data: {
+        action: "BuyConfirm",
+        type: "GLV",
+        poolName: metricData.marketName || "",
+        glvAddress: metricData.glvAddress || "",
+        amount: metricData.glvTokenUsd || 0,
+        isFirstBuy: Boolean(metricData.isFirstBuy),
+      },
+    });
+  }
```

Comment:
I realized that switch / case is not very appropriate here because there could be the same handler for several order types

<img width="510" alt="Screenshot 2024-11-19 at 2 37 57 PM" src="https://github.com/user-attachments/assets/d3aa1924-05f8-4f53-aa33-17bb42e25e78">